compiler/testData/diagnostics/tests/controlFlowAnalysis/doWhileAssignment.fir.kt

```diff
@@ -3,21 +3,21 @@
 fun test() {
     do {
         val s: String
-        s = ""
+        <!VAL_REASSIGNMENT!>s<!> = ""
     } while (s == "")
 }
 
 fun test2() {
     do {
         val s: String
-        s = "1"
-        s = s + "2"
+        <!VAL_REASSIGNMENT!>s<!> = "1"
+        <!VAL_REASSIGNMENT!>s<!> = s + "2"
```

Comment:
These look quite strange to me. The 2nd assignment is definitely `VAL_REASSIGNMENT`, but because of that, the loop ending condition, `s == "1"`, holds and the 1st assignment is regarded as valid assignment at old FE.

Reply:
This (the first one) too.

Reply:
`// FIR_IDENTICAL` !